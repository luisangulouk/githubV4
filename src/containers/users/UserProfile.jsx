/* eslint-disable no-console */
import React, {useState} from 'react';
import FetchReposActivity from '../repositories';
import Style from './users.scss';

const UserProfile = ({name, login, avatarUrl, url, bio, repositories}) => {
  const [activity, setActivity] = useState(false);
  return (
    <article className={`row ${Style.cardWrapper}`}>
      <div className="col-sm-3">
        <div className={Style.cardGithubProfile}>
          <img className={Style.cardImg} src={avatarUrl} alt={name} />
        </div>
      </div>
      <div className="col-sm-9">
        <div className={Style.cardGithubProfile}>
          <h5 className={Style.cardDevName}>{name}</h5>
          <a href={url} className={Style.cardGithubLink}>{login}</a>
          <p className={Style.cardRepos}>Repos: {repositories.totalCount}</p>
          <p className={Style.cardBio}>{bio}</p>
          <button
            type="button"
            className="btn btn-secondary"
            onClick={() => setActivity(true)}
          >Show Activity</button>
          {activity && <FetchReposActivity login={login}/>}
        </div>
      </div>
    </article>
  );
};

export default UserProfile;
