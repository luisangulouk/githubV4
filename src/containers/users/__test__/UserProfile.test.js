/* eslint-disable no-console */
import React from 'react';
import renderer from 'react-test-renderer';

import UserProfile from '../UserProfile';

describe('User List Component', () => {
  const mockProfiles = {
    nodes: [{
      avatarUrl: 'https://avatars2.githubusercontent.com/u/20256970?u=6420f0aa27a29caec0c6ca6aa13af64803923652&v=4',
      bio: 'Its been a while since vanilla was the only flavour...',
      createdAt: '2016-07-02T14:23:00Z',
      id: 'MDQ6VXNlcjIwMjU2OTcw',
      login: 'luisangulouk',
      name: 'Luis Miguel Angulo',
      repositories: {
        totalCount: 25,
        __typename: 'RepositoryConnection'
      },
      url: 'https://github.com/luisangulouk'
    }],
    pageInfo: {
      endCursor: 'Y3Vyc29yOjk=',
      hasNextPage: false
    }
  };

  it('should match snapshot', () => {

    const node = mockProfiles.nodes[0];
    const component = renderer.create(<UserProfile {...node} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
