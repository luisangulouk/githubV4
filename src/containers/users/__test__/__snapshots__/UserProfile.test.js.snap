// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User List Component should match snapshot 1`] = `
<article
  className="row cardWrapper"
>
  <div
    className="col-sm-3"
  >
    <div
      className="cardGithubProfile"
    >
      <img
        alt="Luis Miguel Angulo"
        className="cardImg"
        src="https://avatars2.githubusercontent.com/u/20256970?u=6420f0aa27a29caec0c6ca6aa13af64803923652&v=4"
      />
    </div>
  </div>
  <div
    className="col-sm-9"
  >
    <div
      className="cardGithubProfile"
    >
      <h5
        className="cardDevName"
      >
        Luis Miguel Angulo
      </h5>
      <div
        className="cardCreatedAt"
      >
        Github user since: 
        <time
          dateTime={"2016-07-02T14:23:00.000Z"}
        >
          2016
        </time>
      </div>
      <p
        className="cardBio"
      >
        Its been a while since vanilla was the only flavour...
      </p>
      <a
        className="cardGithubLink"
        href="https://github.com/luisangulouk"
      >
        luisangulouk
      </a>
      <br />
      <span
        className="cardRepos"
        title="Repositories found"
      >
        25
      </span>
      <br />
      <div
        className="cardActivity"
      >
        <button
          className="btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Show
           Activity
        </button>
      </div>
    </div>
  </div>
</article>
`;
